TicTacToe/TicTacToe_hardcore.py

```diff
@@ -0,0 +1,183 @@
+# Additions in hardcore mode:
+# Customizable board size
+# Option to play new game
+# Win counter
+
+board_size = 3
+board = []
+
+player_X = "X"
+player_O = "O"
+current_player = player_X
+playing = True
+wins = {
+    player_X: 0,
+    player_O: 0
+}
+
+# returns if index is the last in a list
+def not_last(index, list):
+    if index == len(list) - 1:
+        return False
+    return True
+
+# displays board
+def print_board():
+    column_indexes_print = "\n  "
+    for column_index in range(len(board[0])):
+        column_indexes_print += f"{column_index}   "
+    print(column_indexes_print)
+
+    for row_index in range(len(board)):
+        row_print = f"{row_index}"
+        for column_index in range(len(board[row_index])):
+            row_print += f" {board[row_index][column_index]} "
+            if not_last(column_index, board[row_index]):
+                row_print += "|"
+        print(row_print)
+
+        if not_last(row_index, board):
+            row_border_print = " "
+            for column_index in range(len(board[row_index])):
+                row_border_print += "---"
+                if not_last(column_index, board[row_index]):
+                    row_border_print += "+"
+            print(row_border_print)
+
+# returns if all elements in a list are the same
+def all_same(list):
+    for item in list:
+        if item != list[0] or item == " ":
+            return False
+
+    return True
+
+# checks if someone wins
+def win():
+    # checks horizontally
+    for row in board:
+        if all_same(row):
+            return True
+
+    # checks vertically
+    for column in range(len(board[0])):
+        check = []
+        for row in board:
+            check.append(row[column])
+        if all_same(check):
+            return True
+
+    # checks diagonally (\)
+    check = []
+    for index in range(len(board)):
+        check.append(board[index][index])
+    if all_same(check):
+        return True
+
+    #checks diagonally (/)
```

Comment:
We didn't teach the zip function yet (though it is quite elegant 😍), I think it might be better just use a loop here (i.e., `board[index][len(board)-index-1]` or whatever other option that doesn't use zip